docs/proposals/compute-resource-metrics-api.md

```diff
@@ -80,16 +143,73 @@ namespace or service aggregation, metrics at those levels could also be
 exposed taking advantage of the fact that Heapster already does aggregation
 and metrics for them.
 
-Initially, this proposal included raw metrics alongside the derived metrics.
-After revising the use cases, it was clear that raw metrics could be left out
-of this proposal. They can be dealt with in a separate proposal, exposing them
-in the Kubelet API via proper versioned endpoints for Heapster to poll
-periodically.
+## Schema
+
+Types are colocated with other API groups in `/pkg/apis/metrics`, and follow api
+groups conventions there.
+
+```go
+// Raw metrics are only available through the kubelet API.
+type RawNode struct {
+  TypeMeta
+  ObjectMeta              // Should include node name
+  Machine ContainerMetrics
+  SystemContainers []ContainerMetrics
+}
+type RawPod struct {
+  TypeMeta
+  ObjectMeta              // Should include pod name
```

Comment:
Yes. Its either that or we can pre-define an infrastructure container and
associate all pod level stats to that container. Even in that case, volume
fs usage cannot belong to any containers.

On Thu, Oct 29, 2015 at 2:29 PM, Tim St. Clair <notifications@github.com>
wrote:

> In docs/proposals/compute-resource-metrics-api.md
> <https://github.com/kubernetes/kubernetes/pull/15862#discussion_r43446885>
> :
>
> > +## Schema
> > +
> > +Types are colocated with other API groups in `/pkg/apis/metrics`, and follow api
> > +groups conventions there.
> > +
> > +```go
> > +// Raw metrics are only available through the kubelet API.
> > +type RawNode struct {
> > +  TypeMeta
> > +  ObjectMeta              // Should include node name
> > +  Machine ContainerMetrics
> > +  SystemContainers []ContainerMetrics
> > +}
> > +type RawPod struct {
> > +  TypeMeta
> > +  ObjectMeta              // Should include pod name
>
> Would we move those stats out of ContainerStats
> <https://github.com/timstclair/cadvisor/blob/master/info/v2/container.go#L105>
> and into RawPod, add additional NetworkStats into the RawPod, or add a new
> Network stats type?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15862/files#r43446885>.
>